crates/re_types_builder/src/codegen/rust/deserializer.rs

```diff
@@ -210,15 +210,18 @@ pub fn quote_arrow_deserializer(
             }
 
             DataType::Union(_, _, arrow2::datatypes::UnionMode::Sparse) => {
-                // We use sparse unions for enums, which means only 8 bits is required for each field,
+                // We use sparse arrow unions for c-style enums, which means only 8 bits is required for each field,
                 // and nulls are encoded with a special 0-index `_null_markers` variant.
+                //
+                // TODO(emilk): verify that the incoming datatype has the same variant names
+                // in the same order as we expect.
```

Comment:
On the contrary, I think that's one of the main mistakes we've made in the past and that resulted in major performance bottlenecks all over: checking that the data coming over the wire matches exactly what we expect, instead of just trying to interpret it as if it did -- no matter what.

We've already made that change in some places (e.g. vec[2|3] iirc), with gigantic gains.
I'd want to do more of that in the future (removing checks), instead of adding more.

Reply:
https://github.com/rerun-io/rerun/issues/5305

Reply:
Good point - the check shouldn't be here, but during ingestion for backwards compatibility reasons.